src/realm/object-store/results.hpp

```diff
@@ -156,6 +156,9 @@ class Results {
     // Create a new Results by removing duplicates based on the specified key paths.
     Results distinct(std::vector<std::string> const& keypaths) const REQUIRES(!m_mutex);
 
+    // Create a new Results by filtering using a user supplied function.
```

Comment:
We should add a comment here that the user supplied function can be called from any thread, so it has to be a pure function or at least thread safe.

Reply:
Done